assets/js/modules/adsense/components/dashboard/AdBlockingRecoveryWidget.test.js

```diff
@@ -0,0 +1,303 @@
+/**
+ * AdBlockingRecoveryWidget component tests.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import AdBlockingRecoveryWidget from './AdBlockingRecoveryWidget';
+import {
+	act,
+	fireEvent,
+	render,
+	createTestRegistry,
+	provideSiteInfo,
+	provideUserAuthentication,
+	provideModules,
+	unsubscribeFromAll,
+} from '../../../../../../tests/js/test-utils';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	MODULES_ADSENSE,
+} from '../../datastore/constants';
+import {
+	ACCOUNT_STATUS_PENDING,
+	ACCOUNT_STATUS_READY,
+	SITE_STATUS_ADDED,
+	SITE_STATUS_READY,
+} from '../../util';
+
+describe( 'AdSenseConnectCTA', () => {
+	let registry;
+	const validSettings = {
+		accountID: 'pub-12345678',
+		clientID: 'ca-pub-12345678',
+		useSnippet: false,
+		accountStatus: ACCOUNT_STATUS_READY,
+		siteStatus: SITE_STATUS_READY,
+		adBlockingRecoverySetupStatus:
+			AD_BLOCKING_RECOVERY_SETUP_STATUS_SETUP_CONFIRMED,
+	};
+
```

Comment:
Similarly to the `SearchFunnelWidget/Overview` test, we can make use of `getWidgetComponentProps()` to retrieve "real" `Widget` and `WidgetNull` components and keep a reference to them here rather than stubbing them further down in the tests.

```suggestion
	const { Widget, WidgetNull } = getWidgetComponentProps( 'adBlockingRecovery' );
```

For example, see here:

https://github.com/google/site-kit-wp/blob/d881aa20e4fbf189a6754aa0930259b5aaee5aa9/assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/Overview.test.js#L44

Reply:
Updated.